dcpy/connectors/edm/publishing.py

```diff
@@ -202,22 +204,53 @@ def publish(
     keep_draft: bool = True,
     max_files: int = s3.MAX_FILE_COUNT,
     latest: bool = False,
+    overwrite_if_exists: bool = False,
 ) -> None:
     """Publishes a specific draft build of a data product
     By default, keeps draft output folder"""
     if version is None:
         version = get_version(draft_key)
+
+    version_already_published = version in get_published_versions(
+        product=draft_key.product
+    )
+    # Bump version if input version already exists and overwrite isn't allowed
+    if version_already_published and not overwrite_if_exists:
+        version = versions.bump(
+            previous_version=version, bump_type=versions.VersionSubType.patch, bump_by=1
+        )
+
     source = draft_key.path + "/"
     target = f"{draft_key.product}/publish/{version}/"
     s3.copy_folder(BUCKET, source, target, acl, max_files=max_files)
+
+    # if current version comes after 'latest' version or there are no files in 'latest' folder,
+    # update 'latest' folder
     if latest:
-        s3.copy_folder(
-            BUCKET,
-            source,
-            f"{draft_key.product}/publish/latest/",
-            acl,
-            max_files=max_files,
-        )
+        try:
+            latest_version = get_latest_version(draft_key.product)
+
+            # Both latest_version and version are expected to be of same version schema
+            after_latest_version = versions.is_newer(
+                version_1=version, version_2=latest_version
+            )
+        except ClientError:
```

Comment:
Is this what we expect to happen when there are no existing versions? (if so, we really should handle this exception in `get_latest_version`)